### PR TITLE
One click unsubscribe url

### PR DESCRIPTION
**Pull request by BlessedDev**
This PR updates all our API client documents and the generic API document to detail the addition of the `one_click_unsubscribe_url` parameter.

### CODE REVIEW

File: source/documentation/_api_docs.md [hunk 1]
```diff
@@ -242,6 +242,24 @@ An identifier you can create if necessary. This reference identifies a single no
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```json
+"one_click_unsubscribe_url": "https://example.com/unsubscribe.html?opaque=123456789"
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/_api_docs.md:257]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_java.md [hunk 2]
```diff
@@ -212,6 +212,24 @@ A unique identifier you create. This reference identifies a single unique notifi
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```java
+URI oneClickUnsubscribeURL = 'https://example.com/unsubscribe.html?opaque=123456789'
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_java.md:227]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_net.md [hunk 3]
```diff
@@ -268,6 +268,24 @@ string reference: "STRING";
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```csharp
+string oneClickUnsubscribeURL : 'https://example.com/unsubscribe.html?opaque=123456789';
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_net.md:283]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_node.md [hunk 4]
```diff
@@ -233,6 +234,24 @@ A unique identifier you create. This reference identifies a single unique notifi
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```javascript
+oneClickUnsubscribeURL = 'https://example.com/unsubscribe.html?opaque=123456789'
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_node.md:249]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_php.md [hunk 5]
```diff
@@ -214,9 +215,26 @@ A unique identifier you can create if necessary. This reference identifies a sin
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```php
+$one_click_unsubscribe_url = 'https://example.com/unsubscribe.html?opaque=123456789'
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_php.md:232]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_python.md [hunk 6]
```diff
@@ -195,6 +195,24 @@ reference='STRING', # optional string - identifies notification(s)
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```python
+one_click_unsubscribe_url = 'https://example.com/unsubscribe.html?opaque=123456789'
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_python.md:210]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_ruby.md [hunk 7]
```diff
@@ -203,6 +203,24 @@ reference: "your_reference_string"
...
+
+This is an optional argument that is only required for subscription emails.
+
+Read our Using Notify page for more information about [unsubscribe links](https://www.notifications.service.gov.uk/using-notify/unsubscribe-links). 
+
+```ruby
+one_click_unsubscribe_url: "https://example.com/unsubscribe.html?opaque=123456789"
+```
+
+The one-click unsubscribe URL must respond to an empty POST request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```
Comment (quis) [inline source/documentation/client_docs/_ruby.md:218]:
```suggestion
The one-click unsubscribe URL must respond to an empty `POST` request by unsubscribing the user from your emails. You can include query parameters to help you identify the user.
```

File: source/documentation/client_docs/_php.md [hunk 8]
```diff
@@ -214,9 +215,26 @@ A unique identifier you can create if necessary. This reference identifies a sin
...
+
+Your unsubscribe URL and response must comply with the guidance specified in [Section 3.1 of IETF RFC 8058](https://www.rfcreader.com/#rfc8058_line139).
+
+You can leave out this argument if the email being sent is not a subscription email.
+
```
Comment (quis) [inline source/documentation/client_docs/_php.md:237]: I don’t think we need this – it duplicates the second paragraph
```suggestion
```

File: source/documentation/client_docs/_php.md [hunk 9]
```diff
@@ -171,7 +171,8 @@ try {
             'dob'  => '12 July 1968'
         ],
         'unique_ref123',
-        '862bfaaf-9f89-43dd-aafa-2868ce2926a9'
+        '862bfaaf-9f89-43dd-aafa-2868ce2926a9',
+        'https://example.com/unsubscribe.html?opaque=123456789'
```
Comment (quis) [inline source/documentation/client_docs/_php.md:175]: I don’t think we should include this in the default example – the more common use case will be to not have an unsubscribe URL
```suggestion
        '862bfaaf-9f89-43dd-aafa-2868ce2926a9'
```

File: source/documentation/_api_docs.md [hunk 10]
```diff
@@ -242,6 +242,24 @@ An identifier you can create if necessary. This reference identifies a single no
 ```
 You can leave out this argument if you do not have a reference.
 
+##### one_click_unsubscribe_url (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/_api_docs.md:250]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_java.md [hunk 11]
```diff
@@ -212,6 +212,24 @@ A unique identifier you create. This reference identifies a single unique notifi
 String reference='STRING';
 ```
 
+##### oneClickUnsubscribeURL (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_java.md:220]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_net.md [hunk 12]
```diff
@@ -268,6 +268,24 @@ string reference: "STRING";
 ```
 You can leave out this argument if you do not have a reference.
 
+##### oneClickUnsubscribeURL (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_net.md:276]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_node.md [hunk 13]
```diff
@@ -233,6 +234,24 @@ A unique identifier you create. This reference identifies a single unique notifi
 "your_reference_here"
 ```
 
+##### oneClickUnsubscribeURL (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_node.md:242]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_php.md [hunk 14]
```diff
@@ -214,9 +215,26 @@ A unique identifier you can create if necessary. This reference identifies a sin
...
 ```
-
 You can leave out this argument if you do not have a reference.
 
+##### one_click_unsubscribe_url (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_php.md:225]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_python.md [hunk 15]
```diff
@@ -195,6 +195,24 @@ reference='STRING', # optional string - identifies notification(s)
 
 You can leave out this argument if you do not have a reference.
 
+##### one_click_unsubscribe_url (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_python.md:203]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

File: source/documentation/client_docs/_ruby.md [hunk 16]
```diff
@@ -203,6 +203,24 @@ reference: "your_reference_string"
 
 You can leave out this argument if you do not have a reference.
 
+##### one_click_unsubscribe_url (optional)
+
+The one-click unsubscribe URL will be added to the headers of your email. Email clients will use it to add an unsubscribe button.
+
+This is an optional argument that is only required for subscription emails.
+
```
Comment (quis) [inline source/documentation/client_docs/_ruby.md:211]: Our pattern for optional arguments seems to be explaining why they are optional in the final paragraph – this is duplicative
```suggestion
```

**Review by quis — CHANGES_REQUESTED** [review]
Only some of the clients have the _Response_ section of `get_notification` updated to include `one_click_unsubscribe_url`. We should be consistent about this across the clients

File: source/documentation/client_docs/_php.md (hunk 9 above)
Comment (BlessedDev) [reply source/documentation/client_docs/_php.md:175]: In this case I followed the pattern where `reference` was added in the example although it is an optional argument. Also it is a good way to draw attention to the new feature.

**Review by leohemsted — APPROVED** [review]

**Pull request merged**